includes/tax-functions.php

```diff
@@ -442,3 +410,60 @@ function edd_download_is_tax_exclusive( $download_id = 0 ) {
 
 	return (Bool) apply_filters( 'edd_download_is_tax_exclusive', $ret, $download_id );
 }
+
+/**
+ * Gets the tax rate object from the database for a given country / region.
+ * Used in `edd_get_tax_rate`, `edd_build_order`, `edd_add_manual_order`.
+ *
+ * @param array $args {
+ *     Country and, optionally, region to get the tax rate for.
+ *
+ *     @type string $country Required - country to check.
+ *     @type string $region  Optional - check a specific region within the country.
+ * }
+ * @return object|boolean
+ *
+ * @since 3.0
+ */
+function edd_get_tax_rate_for_location( $args ) {
+
+	$rate = false;
+	if ( empty( $args['country'] ) ) {
+		return $rate;
+	}
+
+	// Fetch all the tax rates from the database.
+	// The region is not passed in deliberately in order to check for country-wide tax rates.
+	$tax_rates = edd_get_tax_rates(
+		array(
+			'name'   => $args['country'],
+			'status' => 'active',
+		),
+		OBJECT
+	);
+
+	if ( empty( $tax_rates ) ) {
+		return $rate;
+	}
+
+	foreach ( $tax_rates as $tax_rate ) {
+
+		// Regional tax rate.
+		if ( ! empty( $args['region'] ) && ! empty( $tax_rate->description ) ) {
+			if ( strtolower( $args['region'] ) !== strtolower( $tax_rate->description ) ) {
+				continue;
+			}
+
+			$regional_rate = $tax_rate->amount;
+
+			if ( ! empty( $regional_rate ) ) {
+				return $tax_rate;
+			}
+		} elseif ( 'country' === $tax_rate->scope ) {
+			// Countrywide tax rate.
+			$rate = $tax_rate;
+		}
+	}
+
+	return $rate;
```

Comment:
The original `edd_get_tax_rate()` function used `number_format()` internally to round to 4 decimal places, but I see this new function does not do that. Just noting in case this was unintentional.

Reply:
This returns the entire object, not just the rate, but I should double check where the `amount` is used in the order creation process.